server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2791,16 +2792,60 @@ public InputStream maybeRateLimitSnapshots(InputStream stream) {
 
     @Override
     public IndexShardSnapshotStatus getShardSnapshotStatus(SnapshotId snapshotId, IndexId indexId, ShardId shardId) {
-        BlobStoreIndexShardSnapshot snapshot = loadShardSnapshot(shardContainer(indexId, shardId), snapshotId);
-        return IndexShardSnapshotStatus.newDone(
-            snapshot.startTime(),
-            snapshot.time(),
-            snapshot.incrementalFileCount(),
-            snapshot.totalFileCount(),
-            snapshot.incrementalSize(),
-            snapshot.totalSize(),
-            null
-        ); // Not adding a real generation here as it doesn't matter to callers
+        final StepListener<RepositoryData> repoDataListener = new StepListener<>();
+        getRepositoryData(repoDataListener);
+        AtomicBoolean isRemoteIndexShard = new AtomicBoolean(false);
+        repoDataListener.whenComplete(repositoryData -> {
+            // Check if Index meta-N file is written, then read that file to check if this is a remote store enabled index or not. If the
+            // file is absent, get the info from cluster state.
+            BlobContainer indexContainer = indexContainer(indexId);
+            String indexMetaFile = METADATA_PREFIX
+                + repositoryData.indexMetaDataGenerations().indexMetaBlobId(snapshotId, indexId)
+                + ".dat";
+            final IndexMetadata indexMetadata;
+            Map<String, BlobMetadata> indexMetaFiles = indexContainer.listBlobsByPrefix(METADATA_PREFIX);
+            if (indexMetaFiles.containsKey(indexMetaFile)) {
+                indexMetadata = getSnapshotIndexMetaData(repositoryData, snapshotId, indexId);
+            } else {
+                indexMetadata = clusterService.state().metadata().index(indexId.getName());
+            }
+            isRemoteIndexShard.set(
+                indexMetadata.getSettings().getAsBoolean(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, false)
+                    && Boolean.TRUE.equals(getSnapshotInfo(snapshotId).isRemoteStoreIndexShallowCopyEnabled())
+            );
+        },
```

Comment:
Do we need to explicitly assert the presence of the metadata file for remote index

Reply:
No, I think the check should suffice here. This check is meant for situations where a snapshot is being currently processed and the index meta-N file is yet to be written, so we will check cluster state to know if the index has remote store enabled or not.

Reply:
The concern is there is no single source of truth.

Reply:
Outdated change not being used.